carcara/src/checker/rules/strings.rs

```diff
@@ -6,6 +6,19 @@ use crate::{ast::*, checker::error::CheckerError};
 use rug::Integer;
 use std::{cmp, time::Duration};
 
+/// A function that takes an `Rc<Term>` and returns a vector corresponding to
+/// the flat form of that term.
+///
+/// In this flat form, all `str.++` applications are dissolved and every
+/// argument of those applications is inserted into the vector (e.g., `(str.++
+/// a (str.++ b c))` would lead to `[a, b, c]`, where `a`, `b` e `c` são
+/// arbitrary String terms). Furthermore, all String constants are broken
```

Comment:
```suggestion
/// a (str.++ b c))` would lead to `[a, b, c]`, where `a`, `b` and `c` are
/// arbitrary String terms). Furthermore, all String constants are broken
```